app/javascript/dashboard/routes/dashboard/conversation/contact/ContactInfo.vue

```diff
@@ -291,7 +291,17 @@ export default {
         this.$emit('panel-close');
         this.showAlert(this.$t('DELETE_CONTACT.API.SUCCESS_MESSAGE'));
         if (this.$route.name !== 'contacts_dashboard') {
-          this.$router.push({ name: 'contacts_dashboard' });
+          if (this.$route.name === 'conversation_through_inbox') {
+            // We want to stay on the inbox dashboard if we are deleting a contact from the inbox conversation
+            this.$router.push({
+              name: 'inbox_dashboard',
+              params: {
+                inboxId: this.$route.params.inbox_id,
+              },
+            });
+          } else {
+            this.$router.push({ name: 'contacts_dashboard' });
+          }
```

Comment:
You can move the logic outside the if condition

```js
if (this.$route.name === 'conversation_through_inbox') {
   // route to inbox_dashbord
} else if (this.$route.name !== 'contacts_dashboard') {
  // route to contacts_dashboard
}
```

Reply:
@scmmishra Done :)